packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -272,7 +267,8 @@ export abstract class FluidDataStoreContext
 		return this._contextDeltaManagerProxy;
 	}
 
-	public isReadOnly = (): boolean => this.parentContext.isReadOnly();
+	private isStagingReadOnly: boolean = false;
+	public isReadOnly = (): boolean => this.isStagingReadOnly || this.parentContext.isReadOnly();
```

Comment:
The `isStagingReadOnly` member probably works better as the less-complex `isStagingMode`, and then compute whether that implies stagingReadOnly inside `isReadOnly`

```suggestion
	private isStagingMode: boolean = false;
	public isReadOnly = (): boolean => (this.isStagingMode && this.channel?.policies?.readonlyInStagingMode !== false) || this.parentContext.isReadOnly();
```

And correspondingly changes in `notifyStagingMode()` below